### ISSUE TITLE
[Bug]: [Pack] Multiple NU1102 errors occur when running command “dotnet pack --runtime win10-x64” 

### ISSUE THREAD

**Issue by v-luzh** [NuGet/Home#11313]
### NuGet Product Used

dotnet.exe

### Product Version

Dotnet SDK Version: 6.0.0-rc.2.21474.18

### Worked before?

Worked on Dotnet SDK Version:6.0.100-rc.1.21458.32

### Impact

It's more difficult to complete my work

### Repro steps
1.	Launch VS and create a new C# Class Library (.NET 6.0).
2.	Open the Developer Command Prompt and cd to the project path.
3.	Run “`dotnet pack --runtime win10-x64`”.
### Expected
Running command “`dotnet pack --runtime win10-x64`” should be successful.
### Actual
Multiple NU1102 errors  (e.g. “NU1102: Unable to find package Microsoft.NETCore.App.Runtime.win-x64”) occur when running command “`dotnet pack --runtime win10-x64`” as below screenshot.
![image](https://user-images.githubusercontent.com/20632625/137464434-d110455f-60f7-4d40-9647-b7a413b5b5e9.png)
### Notes:
It didn’t repro on Main\31726.35 + NuGet Client Dev\6.0.0.260 on .NET SDK 6.0.100-rc.1.21458.32.

### Verbose Logs

```shell
Error Log:
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Runtime.win-x64 with version (= 6.0.0-rc.2.21474.18)
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102:   - Found 63 version(s) in nuget.org [ Nearest version: 6.0.0-rc.2.21480.5 ]
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102:   - Found 0 version(s) in C:\Program Files\dotnet\library-packs
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102: Unable to find package Microsoft.WindowsDesktop.App.Runtime.win-x64 with version (= 6.0.0-rc.2.21475.3)
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102:   - Found 63 version(s) in nuget.org [ Nearest version: 6.0.0-rc.2.21501.6 ]
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102:   - Found 0 version(s) in C:\Program Files\dotnet\library-packs
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102: Unable to find package Microsoft.AspNetCore.App.Runtime.win-x64 with version (= 6.0.0-rc.2.21475.12)
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102:   - Found 66 version(s) in nuget.org [ Nearest version: 6.0.0-rc.2.21480.10 ]
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages
C:\Users\v-luzh\source\repos\Package002\Package002\Package002.csproj : error NU1102:   - Found 0 version(s) in C:\Program Files\dotnet\library-packs
```

**Comment by zkat** [comment]
@heng-liu do you know how to configure the sources so they work properly here?

**Comment by nkolev92** [comment]
@v-luzh 

When working with dotnet nightlies, you need to configure an additional source, can you please refer to https://github.com/dotnet/installer#installers-and-binaries. 

For example, curently for .NET 6 builds, you need 

```xml
<configuration>
  <packageSources>
    <add key="dotnet6" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json" />
  </packageSources>
</configuration>
```

This would likely change over time, as the major version changes. 

Can you try reproing with the source in question.

**Comment by v-luzh** [comment]
Hi @nkolev92, I still got the similar errors as below after adding the source.
![image](https://user-images.githubusercontent.com/20632625/139034331-197abdb0-3afb-4340-92dc-5c5f9bb1d065.png)

**Comment by nkolev92** [comment]
Thanks for attempting that @v-luzh. 

This is definitely a test case problem, I think we need some deeper analysis to make the test case full proof.

**Comment by heng-liu** [comment]
I tried to repro but failed to install the dotnet version 6.0.0-rc.2.21474.18
`dotnet-install.ps1 -Channel main -Version 6.0.0-rc.2.21474.18`
```
Could not find/download: ".NET Core SDK" with version = 6.0.0-rc.2.21474.18
Refer to: https://aka.ms/dotnet-os-lifecycle for information on .NET Core support
At C:\Users\adminhenli\scripts\PatchedSDK\dotnet-install.ps1:660 char:5
+     throw "Could not find/download: `"$assetName`" with version = $Sp ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Could not find/...ET Core support:String) [], RuntimeException
    + FullyQualifiedErrorId : Could not find/download: ".NET Core SDK" with version = 6.0.0-rc.2.21474.18
Refer to: https://aka.ms/dotnet-os-lifecycle for information on .NET Core support
```
But it's good to install some other version and I was unable to repro with other version for this issue. (I tried 6.0.100-rc.1.21458.32)
May I know how do we determine the testing version of .NET SDK?

**Comment by nkolev92** [comment]
Assigning to hot seat for investigation.

**Comment by v-luzh** [comment]
It is fixed on VS Main\31904.272 + NuGet Client Dev\6.1.0.39. Close this issue now, thanks.

**Comment by nkolev92** [comment]
I wonder if we should tryt to solve this in a way that it doesn't cause so many false negatives. 

cc @aortiz-msft
